src/pcre2_printint.c

```diff
@@ -436,8 +434,6 @@ while (type > 0)
   else if (list_ind == 2) char_list_add = XCL_CHAR_LIST_LOW_32_ADD;
   else char_list_add = XCL_CHAR_LIST_HIGH_32_ADD;
   }
-
-return code + LINK_SIZE;
```

Comment:
How is this working without adding the link size?

Reply:
It does not seem to be wrong. But it can be unused, if the xclass global siye used instead.

Reply:
In that case, I will investigate further why the old code seemed not to be working.

Reply:
Don't spend too much effort on it. An assert could be good, but if it makes the code too complex, it is not worth it. Perhaps the assert could be in the compiler to check that the XCLASS length is filled correctly (no dummy bytes at the end) to reduce possible mistakes in future code. But it can be done later.

Reply:
I've worked it out: your code to advance to the end of the XCL_LIST is correct (my apologies).

The issue was that I didn't realise there is no XCL_END after an XCL_LIST.

In fact - we should just get rid of XCL_END completely. We can do:

```
code_end = code + GET(code, 1);
while (code < code_end) { ... }

// instead of:

while (*code != XCL_END) { ... }
```

It's a waste of a byte to use an XCL_END terminator, when we've already encoded the length of the XCLASS data.

For now, all I have done is add an assertion that the XCL_LIST is the final item in the XCLASS data.

Reply:
This is true. I suspect it was done this way because of the interpreter. If code processing does not become too complex, I am ok to remove the XCL_END.

Reply:
This PR is complete - no changes to XCL_END right now.